include/veriblock/storage/payloads_repository_rocks.hpp

```diff
@@ -42,172 +35,123 @@ struct PayloadsCursorRocks
-    _iterator->Seek(blockHash);
+  void seek(const payloads_id& key) override {
+    std::string id(reinterpret_cast<const char*>(key.data()), key.size());
+    _iterator->Seek(id);
   }
   void seekToLast() override { _iterator->SeekToLast(); }
   bool isValid() const override { return _iterator->Valid(); }
   void next() override { _iterator->Next(); }
   void prev() override { _iterator->Prev(); }
 
-  hash_t key() const override {
+  payloads_id key() const override {
     if (!isValid()) {
       throw std::out_of_range("invalid cursor");
     }
-    auto key = _iterator->key();
-    hash_t blockHash(key.size());
-    for (size_t i = 0; i < blockHash.size(); ++i) {
-      blockHash[i] = key[i];
-    }
 
-    return blockHash;
+    return payloads_id(_iterator->key().ToString());
   }
 
-  stored_payloads_container_t value() const override {
+  stored_payloads_t value() const override {
     if (!isValid()) {
       throw std::out_of_range("invalid cursor");
     }
-    stored_payloads_container_t payloads_container;
 
     auto value = _iterator->value();
 
-    ReadStream readStream(value.data(), value.size());
-    while (readStream.remaining()) {
-      payloads_container.push_back(
-          stored_payloads_t::fromVbkEncoding(readStream));
-    }
-
-    return payloads_container;
+    return stored_payloads_t::fromVbkEncoding(value.ToString());
   }
 
  private:
   std::shared_ptr<rocksdb::DB> _db;
   std::unique_ptr<rocksdb::Iterator> _iterator;
 };
 
-template <typename Block, typename Payloads>
+template <typename Payloads>
 struct PayloadsRepositoryRocks;
 
-template <typename Block, typename Payloads>
-struct PayloadsWriteBatchRocks : public PayloadsWriteBatch<Block, Payloads> {
-  //! block type
-  using block_t = Block;
+template <typename Payloads>
+struct PayloadsWriteBatchRocks : public PayloadsWriteBatch<Payloads> {
   //! stored payloads type
   using stored_payloads_t = Payloads;
-  //! block hash type
-  using hash_t = typename Block::hash_t;
-  //! block height type
-  using height_t = typename Block::height_t;
+  //! payloads id type
+  using payloads_id = typename Payloads::id_t;
+
+  PayloadsWriteBatchRocks(std::shared_ptr<rocksdb::DB> db,
+                          std::shared_ptr<cf_handle_t> payloadsHandle)
+      : _db(std::move(db)), _payloadsHandle(std::move(payloadsHandle)) {}
 
-  using pair = std::pair<hash_t, stored_payloads_t>;
+  void put(const stored_payloads_t& payloads) override {
+    rocksdb::Status s;
 
-  enum class Operation { PUT, REMOVE_BY_HASH };
+    std::string out{};
 
-  ~PayloadsWriteBatchRocks() override = default;
+    auto id = payloads.getId();
+    auto payloadsBytes = payloads.toVbkEncoding();
 
-  PayloadsWriteBatchRocks(
-      PayloadsRepositoryRocks<block_t, stored_payloads_t>* repo)
-      : _repo(repo) {}
+    rocksdb::Slice key(reinterpret_cast<const char*>(id.data()), id.size());
+    rocksdb::Slice value(reinterpret_cast<const char*>(payloadsBytes.data()),
```

Comment:
write a function `rocksdb::Slice makeSlice(vector)` and use it here, instead of reinterpret cast

Reply:
fixed